File: yns-platform/yns-web/types/onboarding.ts
export type UserType = 'high_school' | 'college' | 'recent_grad';

export interface CareerProfileDraft {
  major: string;
  target_field: string;
  target_level: string;
  skills: string[];
  resume_file: File | null;
  has_job_posting: boolean;
  company: string;
  job_description: string;
}

export interface HighSchoolProfileDraft {
  grade: string;
  target_colleges: string[];
  interested_major_or_field: string;
  interview_type: string;
  activities: string;
  resume_file: File | null;
  has_specific_prompt: boolean;
  college_or_program_name: string;
  interview_description: string;
}

export interface OnboardingSubmission {
  name: string;
  user_type: UserType;
  career_profile?: CareerProfileDraft;
  high_school_profile?: HighSchoolProfileDraft;
}

export type FinalOnboardingPayload = OnboardingSubmission;

export interface OnboardingFormValues {
  name: string;
  user_type: UserType | '';
  major: string;
  target_field: string;
  target_level: string;
  skills: string;
  resume_file: File | null;
  has_job_posting: boolean | null;
  company: string;
  job_description: string;
  grade: string;
  target_colleges: string;
  interested_major_or_field: string;
  interview_type: string;
  activities: string;
  has_specific_prompt: boolean | null;
  college_or_program_name: string;
  interview_description: string;
}

File: yns-platform/yns-api/app/schemas/onboarding.py
from typing import Literal

from pydantic import BaseModel, Field

OnboardingUserType = Literal["high_school", "college", "recent_grad"]


class CareerProfileDraft(BaseModel):
    major: str = ""
    target_field: str = ""
    target_level: str = ""
    skills: list[str] = Field(default_factory=list)
    has_job_posting: bool = False
    company: str = ""
    posting_url: str = ""
    job_description: str = ""


class HighSchoolProfileDraft(BaseModel):
    grade: str = ""
    target_colleges: list[str] = Field(default_factory=list)
    interested_major_or_field: str = ""
    interview_type: str = ""
    activities: str = ""
    has_specific_prompt: bool = False
    college_or_program_name: str = ""
    interview_description: str = ""


class OnboardingSubmitRequest(BaseModel):
    name: str
    user_type: OnboardingUserType
    career_profile: CareerProfileDraft | None = None
    high_school_profile: HighSchoolProfileDraft | None = None


class OnboardingSubmitResponse(BaseModel):
    success: bool
    error: str | None = None
